src/main/java/com/flightstats/hub/app/HubMain.java

```diff
@@ -169,14 +176,23 @@ public Server startServer() throws Exception {
     }
 
     private List<Service> getBeforeHealthCheckServices(Injector injector) {
-        List<Service> services = Stream.of(
+
+        final List<Service> services = Stream.of(
                 InfluxdbReporterLifecycle.class,
                 StatsDReporterLifecycle.class)
                 .map(injector::getInstance)
                 .collect(Collectors.toList());
+
         if (storageBackend == StorageBackend.aws) {
             services.add(injector.getInstance(S3WriteQueueLifecycle.class));
+            if (spokeProperties.isTtlEnforced()) {
```

Comment:
I definitely like initializing/adding the lifecycle-type services here vs in the initializer. 👍 Thanks for moving it.